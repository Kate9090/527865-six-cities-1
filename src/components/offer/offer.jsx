import React from "react";
import PropTypes from 'prop-types';

import {connect} from 'react-redux';
import {getActiveOffer, getFavouritesList, getSelectCity} from '../../reducer/user/selectors';

import {ActionCreator} from '../../reducer/user/user';

import {getHotels} from "../../reducer/data/selectors";

import {getStatusAuthorization, getSelectCityNumber} from "../../reducer/user/selectors";

import Header from '../header/header.jsx';
import ReviewList from '../reviews-list/review-list.jsx';
import {Map} from "../map/map.jsx";

import {PlaceCard} from "../place-card/place-card.jsx";

const Offer = (props) => {

  const {offers, favouriteOffers, checkAuthorization, nameCityOnMap,
    offer, sendOfferToFavourite} = props;

  let neighbourOffer = [];

  neighbourOffer = offers.filter((it) => it.city.name === offer.city.name && it.id !== offer.id)
    .slice(0, 3);

  const onCardClick = () => {
    if (favouriteOffers && offer) {
      let i = favouriteOffers.length;
      sendOfferToFavourite(favouriteOffers, offer, i);
    }
  };

  const _renderMap = () => {

    return <Map
      {...props}
      offer={neighbourOffer}
      nameCityOnMap={nameCityOnMap}
      className="offer-map" />;
  };

  const _renderPlaceCardList = () => {
    return <>
      {neighbourOffer.map((it, num) => <PlaceCard {...props} checkAuthorization={checkAuthorization} offer={it} key={`nearPlace-${num}`}/>)}
    </>;
  };

  return <>
    <Header />
    <main className="page__main page__main--property">
      <section className="property">
        <div className="property__gallery-container container">
          <div className="property__gallery">
            {offer.images.map((it, i) => (
              <div className="property__image-wrapper" key={`images-${i}`}>
                <img className="property__image" src={it} alt="Photo studio" />
              </div>
            )).slice(0, 6)}
          </div>
        </div>
        <div className="property__container container">
          <div className="property__wrapper">
            <div className="property__mark">
              <span>{offer.isPremium ? `Premium` : ``}</span>
            </div>
            <div className="property__name-wrapper">
              <h1 className="property__name">
                {offer.title}
              </h1>
              <button onClick={onCardClick} className="property__bookmark-button button" type="button" style={{backgroundImage: `url('/img/icon-bookmark.svg')`, backgroundRepeat: `no-repeat`}}>
                <svg className="property__bookmark-icon" width="31" height="33">
                  <use xlinkHref="#icon-bookmark"></use>
                </svg>
                <span className="visually-hidden">To bookmarks</span>
              </button>
            </div>
            <div className="property__rating rating">
              <div className="property__stars rating__stars">
                <span style={{width: `96%`}}></span>
                <span className="visually-hidden">Rating</span>
              </div>
              <span className="property__rating-value rating__value">{offer.rating}</span>
            </div>
            <ul className="property__features">
              <li className="property__feature property__feature--entire">
                {offer.type}
              </li>
              <li className="property__feature property__feature--bedrooms">
                {offer.bedrooms} Bedrooms
              </li>
              <li className="property__feature property__feature--adults">
                Max {offer.maxAdults} adults
              </li>
            </ul>
            <div className="property__price">
              <b className="property__price-value">{offer.price}</b>
              <span className="property__price-text">&nbsp;night</span>
            </div>
            <div className="property__inside">
              <h2 className="property__inside-title">What&apos;s inside</h2>
              <ul className="property__inside-list">
                {offer.goods.map((it, i) => (
                  <li className="property__inside-item" key={`goods-${i}`}>
                    {it}
                  </li>
                ))}
              </ul>
            </div>
            <div className="property__host">
              <h2 className="property__host-title">Meet the host</h2>
              <div className="property__host-user user">
                <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                  <img className="property__avatar user__avatar" src={`/${offer.host.avatar_url}`} width="74" height="74" alt="Host avatar" />
                </div>
                <span className="property__user-name">
                  {offer.host.name}
                </span>
                <span className="property__user-status">
                  {offer.is_pro ? `Pro` : ``}
                </span>
              </div>
              <div className="property__description">
                <p className="property__text">
                  {offer.description}
                </p>
              </div>
            </div>
            <ReviewList />
          </div>
        </div>
        {_renderMap()}
      </section>
      <div className="container">
        <section className="near-places places">
          <h2 className="near-places__title">Other places in the neighbourhood</h2>
          <div className="near-places__list places__list">
            {_renderPlaceCardList()}
          </div>
        </section>
      </div>
    </main>
  </>;
};

Offer.propTypes = {
  offer: PropTypes.object,
  offers: PropTypes.arrayOf(PropTypes.shape({
    src: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    rating: PropTypes.number,
    name: PropTypes.string,
    location: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
    }).isRequired,
  })).isRequired,
  favouriteOffers: PropTypes.arrayOf(PropTypes.shape({
    src: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    rating: PropTypes.number,
    name: PropTypes.string,
    location: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
    }),
  })).isRequired,
  sendOfferToFavourite: PropTypes.func.isRequired,
  checkAuthorization: PropTypes.bool.isRequired,
  nameCityOnMap: PropTypes.string.isRequired,
};

export {Offer};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  offer: getActiveOffer(state),
  favouriteOffers: getFavouritesList(state),
  offers: getHotels(state),
  checkAuthorization: getStatusAuthorization(state),
  cityOnMap: getSelectCityNumber(state),
  nameCityOnMap: getSelectCity(state),
});

const mapDispatchToProps = (dispatch) => ({
  sendOfferToFavourite: (favouriteOffers, offer, i) => {
    dispatch(ActionCreator.addCardToFavourite(favouriteOffers, offer, i));
  },
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Offer);
