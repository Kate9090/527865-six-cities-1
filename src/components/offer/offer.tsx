import React from 'react';
import PropTypes from 'prop-types';

import {connect} from 'react-redux';
import {getActiveOffer, getFavouritesList, getSelectCity, getReviews} from '../../reducer/user/selectors';

import {ActionCreator} from '../../reducer/user/user';

import {getHotels, getCities} from '../../reducer/data/selectors';

import {getStatusAuthorization, getSelectCityNumber} from '../../reducer/user/selectors';

import Header from '../header/header.jsx';
import ReviewList from '../reviews-list/review-list.jsx';
import {Map} from '../map/map.jsx';

import {PlaceCard} from '../place-card/place-card.jsx';

import {OfferType, Review, FavouriteOfferType} from '../../types';

const OfferParameters = {
  MAX_IMAGES: 6,
  MAX_NEIGHBOURS: 3,
  FULL_WIDTH_IN_PX: 147,
  NUMBER_OF_STARS: 5,
};

interface Props {
  offer: OfferType,
  offers: OfferType[],
  favouriteOffers: FavouriteOfferType[],
  onSendOfferToFavourite: (favouriteOffers: FavouriteOfferType[], offer: FavouriteOfferType, i: number) => void,
  checkAuthorization: boolean,
  nameCityOnMap: string,
  offerCities: string[],
  reviews: Review[],
}

const Offer: React.FunctionComponent<Props> = (props) => {

  const {offers, favouriteOffers, checkAuthorization, nameCityOnMap,
    offer, onSendOfferToFavourite, offerCities, reviews} = props;

  let neighbourOffer = [];

  neighbourOffer = offers.filter((it) => it.city.name === offer.city.name && it.id !== offer.id)
    .slice(0, OfferParameters.MAX_NEIGHBOURS);
  neighbourOffer[OfferParameters.MAX_NEIGHBOURS] = offer;

  let middleReviews = 0;
  reviews.forEach((it, idx) => {
    middleReviews = (middleReviews + it.rating) / (idx + 1);
  });

  const _handleCardClick = () => {
    if (favouriteOffers && offer) {
      let i = favouriteOffers.length;
      onSendOfferToFavourite(favouriteOffers, offer, i);
    }
  };

  const _renderMap = () => {


    return <Map
      {...props}
      offer={neighbourOffer}
      offerCities={offerCities}
      nameCityOnMap={nameCityOnMap}
      activeCard = {offer}
      className="offer-map" />;
  };

  const _renderPlaceCardList = () => {
    return <>
      {neighbourOffer.map((it, num) => <PlaceCard {...props} checkAuthorization={checkAuthorization} offer={it} key={`nearPlace-${num}`}/>).slice(0, 3)}
    </>;
  };

  return <>
    <Header />
    <main className="page__main page__main--property">
      <section className="property">
        <div className="property__gallery-container container">
          <div className="property__gallery">
            {offer.images.map((it, i) => (
              <div className="property__image-wrapper" key={`images-${i}`}>
                <img className="property__image" src={it} alt="Photo studio" />
              </div>
            )).slice(0, OfferParameters.MAX_IMAGES)}
          </div>
        </div>
        <div className="property__container container">
          <div className="property__wrapper">
            <div className="property__mark">
              <span>{offer.isPremium ? `Premium` : ``}</span>
            </div>
            <div className="property__name-wrapper">
              <h1 className="property__name">
                {offer.title}
              </h1>
              <button onClick={_handleCardClick} className="property__bookmark-button button" type="button" style={{backgroundImage: `url('/img/icon-bookmark.svg')`, backgroundRepeat: `no-repeat`, backgroundSize: `cover`}}>
                <svg className="property__bookmark-icon" width="31" height="33">
                  <use xlinkHref="#icon-bookmark"></use>
                </svg>
                <span className="visually-hidden">To bookmarks</span>
              </button>
            </div>
            <div className="property__rating rating">
              <div className="property__stars rating__stars">
                <span style={{width: middleReviews / OfferParameters.NUMBER_OF_STARS * OfferParameters.FULL_WIDTH_IN_PX}}></span>
                <span className="visually-hidden">Rating</span>
              </div>
              <span className="property__rating-value rating__value">{middleReviews}</span>
            </div>
            <ul className="property__features">
              <li className="property__feature property__feature--entire">
                {offer.type}
              </li>
              <li className="property__feature property__feature--bedrooms">
                {offer.bedrooms} Bedrooms
              </li>
              <li className="property__feature property__feature--adults">
                Max {offer.maxAdults} adults
              </li>
            </ul>
            <div className="property__price">
              <b className="property__price-value">{offer.price}</b>
              <span className="property__price-text">&nbsp;night</span>
            </div>
            <div className="property__inside">
              <h2 className="property__inside-title">What&apos;s inside</h2>
              <ul className="property__inside-list">
                {offer.goods.map((it, i) => (
                  <li className="property__inside-item" key={`goods-${i}`}>
                    {it}
                  </li>
                ))}
              </ul>
            </div>
            <div className="property__host">
              <h2 className="property__host-title">Meet the host</h2>
              <div className="property__host-user user">
                <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                  <img className="property__avatar user__avatar" src={`/${offer.host.avatar_url}`} width="74" height="74" alt="Host avatar" />
                </div>
                <span className="property__user-name">
                  {offer.host.name}
                </span>
                <span className="property__user-status">
                  {offer.isPro ? `Pro` : ``}
                </span>
              </div>
              <div className="property__description">
                <p className="property__text">
                  {offer.description}
                </p>
              </div>
            </div>
            <ReviewList />
          </div>
        </div>
        {_renderMap()}
      </section>
      <div className="container">
        <section className="near-places places">
          <h2 className="near-places__title">Other places in the neighbourhood</h2>
          <div className="near-places__list places__list">
            {_renderPlaceCardList()}
          </div>
        </section>
      </div>
    </main>
  </>;
};

Offer.propTypes = {

};

export {Offer};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  offer: getActiveOffer(state),
  favouriteOffers: getFavouritesList(state),
  offers: getHotels(state),
  checkAuthorization: getStatusAuthorization(state),
  cityOnMap: getSelectCityNumber(state),
  nameCityOnMap: getSelectCity(state),
  offerCities: getCities(state),
  reviews: getReviews(state),
});

const mapDispatchToProps = (dispatch) => ({
  onSendOfferToFavourite: (favouriteOffers, offer, i) => {
    dispatch(ActionCreator.addCardToFavourite(favouriteOffers, offer, i));
  },
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Offer);
