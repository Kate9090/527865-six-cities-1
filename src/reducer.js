
const initialState = {
  city: `Amsterdam`,
  offerInCity: require(`./mocks/offers-in-amsterdam`).offerHotelList,
};


const ActionCreator = ({
  'NEW_CITY': (newCity) => ({
    type: `NEW_CITY`,
    payload: newCity,
  }),

  // 'RESET': () => ({
  //   type: `RESET`,
  // }),

});

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `NEW_CITY`:
      return Object.assign({}, state, {
        city: action.payload,
        offerInCity: require(`./mocks/offers-in-${state.city}`).offer,
      });
    // case `RESET`:
    //   return Object.assign({}, state, initialState);
  }

  return state;
};

export {reducer, ActionCreator};
