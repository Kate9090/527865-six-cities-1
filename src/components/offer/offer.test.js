import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import {Offer} from './offer.jsx';

import offerHotelList from '../../mocks/offers-in-amsterdam';
const mock = offerHotelList;

import {Provider} from 'react-redux';
import reducer from '../../reducer/index';
import {createStore} from 'redux';

const store = createStore(
    reducer
);


it(`renders correctly PlaceCard`, () => {
  const tree = renderer
    .create(<BrowserRouter><Provider store={store}><Offer
      offer = {mock[1]}
      offers = {mock}
    /></Provider></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
