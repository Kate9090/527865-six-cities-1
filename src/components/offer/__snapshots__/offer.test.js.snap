// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly PlaceCard 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="main.html"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <a
              className="header__login"
              href="/login"
              onClick={[Function]}
            >
              Sign in
            </a>
          </ul>
        </nav>
      </div>
    </div>
  </header>,
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/18.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/9.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/12.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/8.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/20.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://es31-server.appspot.com/six-cities/static/hotel/11.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful &amp; luxurious apartment at great location
            </h1>
            <button
              className="property__bookmark-button button"
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "96%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            />
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              house
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              8
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              120
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Fridge
              </li>
              <li
                className="property__inside-item"
              >
                Air conditioning
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
              <li
                className="property__inside-item"
              >
                Laptop friendly workspace
              </li>
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Washer
              </li>
              <li
                className="property__inside-item"
              >
                Breakfast
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
              <span
                className="property__user-status"
              >
                
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              />
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                1
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="img/avatar-max.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Max
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "94%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  />
                  <time
                    className="reviews__time"
                    dateTime="2019-04-24"
                  >
                    April 2019
                  </time>
                </div>
              </li>
            </ul>
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="near-places__card place-card"
          >
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="img/room.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €80
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button place-card__bookmark-button--active button"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                >
                  Wood and stone place
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Private room
              </p>
            </div>
          </article>
          <article
            className="near-places__card place-card"
          >
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-02.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €132
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                >
                  Canal View Prinsengracht
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
          <article
            className="near-places__card place-card"
          >
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-03.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €180
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                >
                  Nice, cozy, warm big bed apartment
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>,
]
`;
